plugin.py

```diff
@@ -5,14 +5,73 @@
 from LSP.plugin.core.typing import Optional, Any, List, Dict, Mapping, Callable
 import os
 import sublime
+from urllib.request import urlopen
+from urllib.error import URLError
+import re
+import shutil
 import tempfile
+import tarfile
 
 # TODO: Not part of the public API :(
 from LSP.plugin.core.edit import apply_workspace_edit
 from LSP.plugin.core.edit import parse_workspace_edit
 from LSP.plugin.core.views import location_to_encoded_filename
 
 
+DOWNLOAD_URL = "http://download.eclipse.org/jdtls/snapshots"
+LATEST_SNAPSHOT = None
+SETTINGS_FILENAME = "LSP-jdtls.sublime-settings"
+STORAGE_DIR = 'LSP-jdtls'
+SERVER_DIR = "server"
+DATA_DIR = "data"
+
+
+def fetch_latest_release() -> None:
+    """
+    Fetches the latest release.
+    """
+    global LATEST_SNAPSHOT
+    if not LATEST_SNAPSHOT:
+        try:
+            with urlopen(DOWNLOAD_URL + "/latest.txt") as f:
+                data = f.read().decode('utf-8')
+                version = re.search("jdt-language-server-(.*).tar.gz", data)
+                LATEST_SNAPSHOT = version.group(1)
+        except URLError:
+            pass
+
+
+fetch_latest_release()
```

Comment:
Don't block the UI while downloading files with `urlopen`. Can we do this in `needs_update_or_installation` instead? That classmethod runs in ST's worker thread.

Reply:
Sure